web/src/AddressesDataList.jsx

```diff
@@ -0,0 +1,138 @@
+ */
+
+/*
+ * Based in the previous work done for cockpit-wicked project, see
+ *   - https://github.com/openSUSE/cockpit-wicked
+ *   - https://github.com/openSUSE/cockpit-wicked/blob/master/src/components/AddressesDataList.js
+ */
+
+import React from "react";
+import {
+  Button,
+  DataList,
+  DataListItem,
+  DataListItemRow,
+  DataListItemCells,
+  DataListCell,
+  DataListAction,
+  TextInput,
+  Stack,
+  StackItem,
+  Split,
+  SplitItem
+} from "@patternfly/react-core";
+
+import PlusIcon from "@patternfly/react-icons/dist/js/icons/plus-icon";
+import MinusIcon from "@patternfly/react-icons/dist/js/icons/minus-icon";
+
+let index = 0;
+
+export default function AddressesDataList({
+  addresses: originalAddresses,
+  updateAddresses,
+  allowEmpty = true
+}) {
+  const addresses = originalAddresses.map(addr => {
+    const newAddr = addr;
+    if (!newAddr.id) newAddr.id = index++;
+    return newAddr;
+  });
+
+  const addAddress = () => {
+    addresses.push({ address: "", prefix: "", id: index++ });
+    updateAddresses(addresses);
+  };
+
+  const updateAddress = (id, field, value) => {
+    const address = addresses.find(addr => addr.id === id);
+    address[field] = value;
+    updateAddresses(addresses);
+  };
+
+  const deleteAddress = id => {
+    const addressIdx = addresses.findIndex(addr => addr.id === id);
+    addresses.splice(addressIdx, 1);
+    updateAddresses(addresses);
+  };
+
+  const renderAddress = ({ id, address, prefix }) => {
+    const renderDeleteAction = () => {
+      if (!allowEmpty && addresses.length === 1) return null;
+
+      return (
+        <DataListAction>
+          <Button variant="secondary" className="btn-sm" onClick={() => deleteAddress(id)}>
+            <MinusIcon />
+          </Button>
+        </DataListAction>
+      );
+    };
+
+    const cells = [
+      <DataListCell key={`address-${id}-local`}>
+        <TextInput
+          defaultValue={address}
+          onChange={value => updateAddress(id, "address", value)}
+          placeholder="Ip Address"
+          aria-label="Ip Address"
+        />
+      </DataListCell>,
+      <DataListCell key={`address-${id}-label`}>
+        <TextInput
+          defaultValue={prefix}
+          onChange={value => updateAddress(id, "prefix", value)}
+          placeholder="Prefix length or netmask"
+          aria-label="Prefix length or netmask"
+        />
+      </DataListCell>
+    ];
+
+    return (
+      <DataListItem key={`address-${id}`}>
+        <DataListItemRow>
+          <DataListItemCells dataListCells={cells} />
+          {renderDeleteAction()}
+        </DataListItemRow>
+      </DataListItem>
+    );
+  };
+
+  return (
+    <Stack className="data-list-form" hasGutter>
+      <StackItem>
+        <Split hasGutter>
+          <SplitItem isFilled>Addresses</SplitItem>
+          <SplitItem>
+            <Button variant="primary" className="btn-sm" onClick={() => addAddress()}>
+              <PlusIcon />
```

Comment:
I have tested to change the icon and use test instead.

What do you think? Which you prefer? Why?

| Using icons | Using text (no address yet) | Using text (at least one address already)
|-|-|-|
| ![Using icons](https://user-images.githubusercontent.com/1691872/196482589-d582aa67-e2c1-4865-b08a-555894122bce.png) | ![Using texts 1/2](https://user-images.githubusercontent.com/1691872/196482793-d7e598fe-2710-4130-8124-a4e8cba8f1dd.png) | ![Using texts 2/2](https://user-images.githubusercontent.com/1691872/196482943-01eeeb8d-045a-403e-96e4-09a765b56c23.png) |


```diff
diff --git a/web/src/AddressesDataList.jsx b/web/src/AddressesDataList.jsx
index f83a7b9..baba9f9 100644
--- a/web/src/AddressesDataList.jsx
+++ b/web/src/AddressesDataList.jsx
@@ -41,9 +41,6 @@ import {
   SplitItem
 } from "@patternfly/react-core";
 
-import PlusIcon from "@patternfly/react-icons/dist/js/icons/plus-icon";
-import MinusIcon from "@patternfly/react-icons/dist/js/icons/minus-icon";
-
 let index = 0;
 
 export default function AddressesDataList({
@@ -81,7 +78,7 @@ export default function AddressesDataList({
       return (
         <DataListAction>
           <Button variant="secondary" className="btn-sm" onClick={() => deleteAddress(id)}>
-            <MinusIcon />
+            Remove
           </Button>
         </DataListAction>
       );
@@ -116,6 +113,8 @@ export default function AddressesDataList({
     );
   };
 
+  const newAddressButtonLabel = addresses.length === 0 ? "Add an address" : "Add another address";
+
   return (
     <Stack className="data-list-form" hasGutter>
       <StackItem>
@@ -123,7 +122,7 @@ export default function AddressesDataList({
           <SplitItem isFilled>Addresses</SplitItem>
           <SplitItem>
             <Button variant="primary" className="btn-sm" onClick={() => addAddress()}>
-              <PlusIcon />
+              {newAddressButtonLabel}
             </Button>
           </SplitItem>
         </Split>

Reply:
IMHO, using buttons with text makes the UI is much more clear .

Reply:
Another option is to preserve the icon 
![Screenshot from 2022-10-19 10-54-10](https://user-images.githubusercontent.com/1691872/196659107-a7697d2c-8a20-42cf-b786-b890a4229c1e.png)
while adding the text

Reply:
Personally I prefer without icons. It does not add value to me. I mean, having the icon does not make me to understand the meaning of the action better. BTW, what about using red color for "Remove" ?

Reply:
For me just icons was fine

Reply:
> Personally I prefer without icons. It does not add value to me. I mean, having the icon does not make me to understand the meaning of the action better. 

Again, I completely agree. The experiment of preserving the icons was done to, somehow, keep a certain level of "consistence" with cockpit UI. Nothing else. But from my point of view, texts makes a better UX. The user does not have to think about the goal of that "+" button there, specially when the form contains more than once, similar, widgets.

> [...] what about using red color for "Remove" ?

I have no a strong opinion about that at this moment, but putting it in red could be a bit "aggressive" or disrupting to me. Maybe we can change to red color when the user over the button.